src/envoy/http/mixer/filter.cc

```diff
@@ -165,29 +165,33 @@ FilterTrailersStatus Filter::decodeTrailers(HeaderMap& trailers) {
   return FilterTrailersStatus::Continue;
 }
 
+void Filter::UpdateHeaders(
+    HeaderMap* headers, const ::google::protobuf::RepeatedPtrField<
```

Comment:
per envoy styple guide,  use & instead of raw pointer

Reply:
Filter is caching a pointer `headers_` to the request header map. Any idea why it is a pointer, not a reference?

Reply:
bcs you only have "headers" at decodeHeader(), not at constructor.

Reply:
how should I cast a pointer to a reference here?

Reply:
*headers_  will cast to reference.